src/components/header/Header.tsx

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import cn from 'classnames';
+import { useEffect, useState } from 'react';
+import { postTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { ErrorMessage } from '../../types/ErrorMessage';
+
+interface Title {
+  title : string;
+}
+
+interface Completed {
+  completed: boolean;
+}
+
+type Props = {
+  setTodoList:(todo: Todo[]) => void,
+  todoList: Todo[],
+  setError:(text: string) => void,
+  addTodoToProcesing:(id: number | null) => void,
+  editTodo:(id: number, itemToEdit: Title | Completed) => void
+};
+
+export const Header:React.FC<Props> = ({
+  setTodoList,
+  todoList,
+  setError,
+  editTodo,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [allActiveButton, setAllActiveButton] = useState(false);
+  const todoForProcessing = {
+    title: inputValue,
+    completed: false,
+    userId: 10514,
+    id: 0,
+  };
+
+  useEffect(() => setAllActiveButton(
+    todoList.some(item => !item.completed),
+  ));
```

Comment:
Are you sure you need this useEffect without a dependency array here?

Reply:
logically, it is not needed at all. But its absence gives an error